src/renderer/families/tezos/accountActions.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+import React, { useCallback } from "react";
+import { connect } from "react-redux";
+import { useDelegation } from "@ledgerhq/live-common/lib/families/tezos/bakers";
+import type { Account, AccountLike } from "@ledgerhq/live-common/lib/types";
+import { openModal } from "~/renderer/actions/modals";
+import {
+  SendActionDefault,
+  ReceiveActionDefault,
+} from "~/renderer/screens/account/AccountActionsDefault";
+
+type OwnProps = {
+  account: AccountLike,
+  parentAccount: ?Account,
+  onClick: () => void,
+};
+
+type Props = OwnProps & {
+  openModal: (modal: string, opts?: *) => void,
+};
+
+const SendActionC = ({ account, parentAccount, onClick, openModal }: Props) => {
+  const delegation = useDelegation(account);
+  const sendShouldWarnDelegation = delegation && delegation.sendShouldWarnDelegation;
+
+  const onClickDecorated = useCallback(() => {
+    if (sendShouldWarnDelegation) {
+      openModal("MODAL_SEND", {
+        parentAccount,
+        account,
+        startWithWarning: sendShouldWarnDelegation,
+      });
+    } else {
+      onClick();
+    }
+  }, [sendShouldWarnDelegation, parentAccount, account, openModal, onClick]);
+
+  return (
+    <SendActionDefault onClick={onClickDecorated} account={account} parentAccount={parentAccount} />
+  );
+};
+
+const ReceiveActionC = ({ account, parentAccount, onClick, openModal }: Props) => {
+  const delegation = useDelegation(account);
+  const receiveShouldWarnDelegation = delegation && delegation.receiveShouldWarnDelegation;
+
+  const onClickDecorated = useCallback(() => {
+    if (receiveShouldWarnDelegation) {
+      openModal("MODAL_RECEIVE", {
+        parentAccount,
+        account,
+        startWithWarning: receiveShouldWarnDelegation,
+      });
+    } else {
+      onClick();
+    }
+  }, [receiveShouldWarnDelegation, parentAccount, account, openModal, onClick]);
+
+  return (
+    <ReceiveActionDefault
+      onClick={onClickDecorated}
+      account={account}
+      parentAccount={parentAccount}
+    />
+  );
+};
+
+const SendAction: React$ComponentType<OwnProps> = connect(null, { openModal })(SendActionC);
```

Comment:
Can we agree on a naming convention now that flow wants us to wrap the connected component?
I'm ok with this one, but I used `ConnectedComponentName` on mine and it would be great if we did it consistently

Reply:
good point, to be honest I think most of the time the original component should be `SendAction`, that way it's still nicely displayed in the React inspector, and the result of `connect(SendAction)` name is not really important (it can even just be `module` or `component`)

Reply:
for this specific file, it is a bit annoying because I use the name as property when doing `export default {SendAction,...}` but maybe I would rename the inner part to something else